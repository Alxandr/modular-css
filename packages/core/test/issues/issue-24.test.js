"use strict";

const dedent = require("dedent");
const namer  = require("test-utils/namer.js");

const Processor = require("../../processor.js");

describe("/issues", () => {
    describe("/24", () => {
        it("should be able to compose using a value", async () => {
            const processor = new Processor({
                namer,
            });
            
            const result = await processor.string(
                "fixtures/composition.css",
                dedent(`
                    @value simple: "./simple.css";
                    
                    .a {
                        composes: fooga from simple;
                        background: #000;
                    }
                `)
            );

            expect(result.exports).toMatchSnapshot();

            const output = await processor.output();
            
            expect(output.css).toMatchSnapshot();
        });
    });
});
