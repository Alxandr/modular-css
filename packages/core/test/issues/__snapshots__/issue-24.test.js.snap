// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/issues /24 should be able to compose using a value 1`] = `
Object {
  "a": Array [
    "fooga",
    "a",
  ],
  "simple": "\\"./simple.css\\"",
}
`;

exports[`/issues /24 should be able to compose using a value 2`] = `
"/* fixtures/simple.css */
.fooga {
    color: red;
}
/* fixtures/composition.css */
.a {
    background: #000;
}
"
`;
